dev-libs/ffcall/ffcall-2.4-r2.ebuild

```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit flag-o-matic libtool toolchain-funcs
+
+MY_PV="libffcall-${PV}"
+
+DESCRIPTION="Build foreign function call interfaces in embedded interpreter"
+HOMEPAGE="https://www.gnu.org/software/libffcall/"
+SRC_URI="mirror://gnu/libffcall/${MY_PV}.tar.gz"
+S="${WORKDIR}"/${MY_PV}
+
+# "Ffcall is under GNU GPL. As a special exception, if used in GNUstep
+# or in derivate works of GNUstep, the included parts of ffcall are
+# under GNU LGPL." -ffcall author
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+
+PATCHES=(
+	# bug 842915, drop on next version bump
+	"${FILESDIR}"/${PN}-2.4-vacall-riscv-pic.patch
+	"${FILESDIR}"/${PN}-2.4-slibtool.patch # 776976
+)
+
+src_prepare() {
+	# The build system is a strange mix of autogenerated
+	# files and manual tweaks on top. Uses $CFLAGS / $LDFLAGS randomly.
+	# We are adding them consistently here and a bit over the top:
+	# bugs: #334581
+	local mfi
+	for mfi in {,*/,*/*/,}Makefile.in ; do
+		einfo "Patching '${mfi}'"
+		# usually uses only assembler here, but -march=
+		# and -Wa, are a must to pass here.
+		sed -e 's/$(CC) /&$(CFLAGS) /g' \
+			-i "${mfi}" || die
+	done
+
+	default
+	elibtoolize
+
+	# for same reason above, patch Makefile.in to add -no-integrated-as as shown in upstream commit:
```

Comment:
Looking at the commit which fixes the previous one (https://gitweb.git.savannah.gnu.org/gitweb/?p=libffcall.git;a=commitdiff;h=35692c0a38c4868f42c884728c23833d8d36e83e), I think we should pull that in or add a workaround in the ebuild like: `use arm && tc-is-clang && append-flags -no-integrated-as` (or whatever it is) and remove it with 2.5?

Reply:
We don't have to patch older versions in-tree if something is already fixed in ~arch and even better already fixed in stable. In this case, the only reason 2.4 is still in-tree is we're waiting on sparc to stable 2.5.

In special cases we can backport fixes to older versions but I don't see a need here.